_posts/2018-04-10-personalized-search.md

```diff
@@ -0,0 +1,301 @@
+    }
+
+Finally we are preparing the search response with a subset of the search results, set by offset and rows.
+
+    final List<ScoredAd> searchResult = results.stream()
+           .skip(offset)
+           .limit(rows)
+           .collect(toList());
+    try {
+       addToResponse(rb, reRankNum, searchResult, initialSearchResult);
+    } catch (IOException e) {
+       LOG.error("Could not add recommended items to search result: " + e);
+    }
+    
+    private static void addToResponse(ResponseBuilder rb, int reRankNum, List<ScoredAd> resultsToReturn, DocList docList) throws IOException {
+       DocListAndSet docListAndSet = buildDocListAndSet(reRankNum, resultsToReturn, docList);
+       rb.setResults(docListAndSet);
+    
+       BasicResultContext resultContext = new BasicResultContext(rb);
+       SolrQueryResponse rsp = rb.rsp;
+       rsp.getValues().removeAll("response");
+       rsp.addResponse(resultContext);
+    }
+
+
+## Response times
+We will also look at the response times:
+
+Response times for 95th percentile. Default search is searches without any custom plugins, distribution search is the actual search we are trying to replace with personalization search. The actual timings in the 95th percentile doubles from 225ms to 450ms. While the basic search takes around 80ms.
+
+<figure>
+    <img class="center-block" src="/images/2018-04-10-personalized-search/95th_percentile_latency.png" alt="alt" title="95th percentile latency" />
+    <figcaption style="text-align:center; font-style:italic;">95th percentile latency</figcaption>
+</figure>
+
+
+Lets also look at the 99th percentile for response time. We here removed the default search, because of too much noise, because of low traffic. 
+The personalization search’s 99th percentile is between 60 to 100 ms slower than the search to be replaced.
+
+<figure>
+    <img class="center-block" src="/images/2018-04-10-personalized-search/99th_percentile_latency.png" alt="alt" title="99th percentile latency" />
+    <figcaption style="text-align:center; font-style:italic;">99th percentile latency</figcaption>
+</figure>
+
+## Search examples
+
+Here is a couple of examples of different user doing the same search at the same time, which will result in a completely different order of the ads:
+
+<figure>
+    <img class="center-block" src="/images/2018-04-10-personalized-search/example_with_personalized1.png" alt="alt" title="example 1 personalized search" />
+    <figcaption style="text-align:center; font-style:italic;">Personalized search with a user interested in antiques</figcaption>
+</figure>
+
+<figure>
+    <img class="center-block" src="/images/2018-04-10-personalized-search/example_with_personalized2.png" alt="alt" title="example 2 personalized search" />
+    <figcaption style="text-align:center; font-style:italic;">Personalized search with a user interested in furnitures for children</figcaption>
```

Comment:
in the final view this looks like a heading of the next paragraph. tune to make it look like the caption of the picture....